crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -102,6 +110,103 @@ impl Consensus {
         sha256::Hash::from_slice(leaf_hash.as_slice())
             .expect("parent_hash: Engines shouldn't be Err")
     }
+    /// Check whether a block is valid
+    /// All validation methods is outsorced to this function
+    ///
+    /// Checks:
+    ///     - Merkle root
+    ///     - BIP34
+    ///     - Witness commitment
+    ///     - Block size (inside Transactions validation)
+    ///     - Transactions
+    ///     - prev block hash
+    pub fn validate_block(
+        &self,
+        block: &bitcoin::Block,
+        ancestor: &BlockHeader,
+        height: u32,
+        inputs: HashMap<bitcoin::OutPoint, bitcoin::TxOut>,
+        flags: c_uint,
+        validate_script: bool,
+        chain_time: ChainTime,
+    ) -> Result<(), BlockchainError> {
+        if !block.check_merkle_root() {
+            return Err(BlockchainError::BlockValidation(
+                BlockValidationErrors::BadMerkleRoot,
+            ));
+        }
+        if height >= self.parameters.bip34_activation_height
```

Comment:
Some of those `if`s can be written more idiomatically with a `map_err` followed by a `?` operator.

Reply:
Actually everything that were checking here is returning boolean and we cant call map_err. Even thats not "indiomatic", it has a pattern.

Reply:
Ok, I think it makes sense